operators/src/util/gdal.rs

```diff
@@ -88,3 +93,58 @@ pub fn gdal_open_dataset(path: &Path) -> Result<Dataset> {
 pub fn gdal_open_dataset_ex(path: &Path, dataset_options: DatasetOptions) -> Result<Dataset> {
     Dataset::open_ex(path, dataset_options).context(error::Gdal)
 }
+
+/// Create a `RasterResultDescriptor` for the given `band` and `dataset`. If the raster data type is
+/// unknown, the default is F64 unless it is otherwise specified by `default_data_type`. If the data
+/// type is a complex floating point type, an error is returned
+pub fn raster_descriptor_from_dataset(
+    dataset: &Dataset,
+    band: isize,
+    default_data_type: Option<RasterDataType>,
+) -> Result<RasterResultDescriptor> {
+    let rasterband = &dataset.rasterband(band)?;
+
+    let spatial_ref: SpatialReference =
+        dataset.spatial_ref()?.try_into().context(error::DataType)?;
+
+    let data_type = match rasterband.band_type() {
+        GDALDataType::GDT_Byte => RasterDataType::U8,
+        GDALDataType::GDT_UInt16 => RasterDataType::U16,
+        GDALDataType::GDT_Int16 => RasterDataType::I16,
+        GDALDataType::GDT_UInt32 => RasterDataType::U32,
+        GDALDataType::GDT_Int32 => RasterDataType::I32,
+        GDALDataType::GDT_Float32 => RasterDataType::F32,
+        GDALDataType::GDT_Float64 => RasterDataType::F64,
+        GDALDataType::GDT_Unknown => default_data_type.unwrap_or(RasterDataType::F64),
+        _ => return Err(Error::GdalRasterDataTypeNotSupported),
+    };
+
+    Ok(RasterResultDescriptor {
+        data_type,
+        spatial_reference: spatial_ref.into(),
+        measurement: Measurement::Unitless,
+        no_data_value: rasterband.no_data_value(),
+    })
+}
+
+/// Create `GdalDatasetParameters` from the infos in the given `dataset` for the given `band`.
+/// `path` is the location of the actual data.
+pub fn gdal_parameters_from_dataset(
+    dataset: &Dataset,
+    band: isize,
+    path: &Path,
+) -> Result<GdalDatasetParameters> {
+    let rasterband = &dataset.rasterband(band)?;
+
+    Ok(GdalDatasetParameters {
+        file_path: PathBuf::from(path),
+        rasterband_channel: usize::try_from(band).unwrap_or(0), // TODO: is the band in the metadata Dataset the same as in the actual data file?
```

Comment:
What do you mean with metadata dataset?

Reply:
the function gets a dataset, band index and a path, and builds the GdalDatasetParameters  from the info in the dataset and uses the given path as file_path. But the path could point to a different file than the Dataset comes from. So you could use the fourth band of dataset a to build the parameters for opening dataset b which only has one band.

Reply:
Is it worth it checking this or do we just get an error (no panic) then?

Reply:
maybe we could introduce another parameter for the band. Then the caller can decide which band to take for creating the Loading Info from and which one to use to load the data?

Reply:
e49700a325e17c2a5dec1dfee35beadc4f0297ca